src/Test/QuickCheck/LCG.purs

```diff
@@ -18,22 +18,39 @@ import qualified Data.Maybe.Unsafe as U
 
 type Seed = Int
 
--- | A magic constant for the linear congruential generator
+-- | The *multiplier*: a magic constant for the linear congruential generator
 lcgM :: Int
-lcgM = 1103515245
+lcgM = 48271
 
--- | A magic constant for the linear congruential generator
+-- | The *increment*: a magic constant for the linear congruential generator
 lcgC :: Int
-lcgC = 12345
+lcgC = 0
 
--- | A magic constant for the linear congruential generator
+-- | The *modulus*: a magic constant for the linear congruential generator
 lcgN :: Int
-lcgN = one `shl` 30
+lcgN = 1 `shl` 31 - 1
```

Comment:
There is an overflow here during `shl`, but the `- 1` puts the result back in the valid range. Not sure if that's something to be concerned about, but `top` might be simpler ?